sfkit/sidecar/server.py

```diff
@@ -0,0 +1,73 @@
+import json
+import os
+import select
+import socket
+import subprocess
+from threading import Thread
+
+from sfkit.sidecar.utils import get_sock_path
+
+
+def handle_client(client: socket.socket):
+    try:
+        while True:
+            data = client.recv(1024)
+            if not data:
+                break
+            try:
+                request = json.loads(data.decode("utf-8"))
+            except json.JSONDecodeError:
+                client.sendall("Invalid JSON format".encode("utf-8"))
+                continue
+
+            study_id = request.get("study_id", "")
+            data_path = request.get("data_path", "")
+
+            client.sendall("Received request".encode("utf-8"))
+
+            command = ["env", "PYTHONUNBUFFERED=x", "sfkit", "all"]
+            if study_id:
+                command.extend(["--study_id", study_id])
+            if data_path:
+                command.extend(["--data_path", data_path])
+
+            try:
+                with subprocess.Popen(
+                    command,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    text=True,
+                    bufsize=1,
+                ) as process:
+                    while process.poll() is None:
+                        rlist, _, _ = select.select([process.stdout, process.stderr], [], [])
+
+                        if not rlist:
+                            process.kill()
+
+                        for stream in rlist:
+                            line = stream.readline().strip()
+                            print(line)
+                            client.sendall(line.encode("utf-8"))
+
+                    process.wait()
```

Comment:
```suggestion
                    while True:
                        line = process.stdout.readline().strip()
                        if line:
                            client.sendall(line.encode("utf-8"))
                            print(line)
                        elif process.poll() is not None:
                            break
```
Simplifying, thanks to combined stderr/stdout above.